test/scripts/hexo/post.js

```diff
@@ -1004,4 +1004,29 @@ describe('Post', () => {
     data.content.trim().should.include('<h1 id="Title-2"><a href="#Title-2" class="headerlink" title="Title 2"></a>Title 2</h1>');
     data.content.trim().should.include('<h1 id="Title-3"><a href="#Title-3" class="headerlink" title="Title 3"></a>Title 3</h1>');
   });
+
+  // test for Issue #3259
+  // https://github.com/hexojs/hexo/issues/3259
+  it('render() - "{{" & "}}" inside inline code', async () => {
+    const content = 'In Go\'s templates, blocks look like this: `{{block "template name" .}} (content) {{end}}`.';
+
+    const data = await post.render(null, {
+      content,
+      engine: 'markdown'
+    });
+
+    data.content.trim().should.eql('<p>In Go’s templates, blocks look like this: <code>&amp;#123;&amp;#123;block &quot;template name&quot; .&amp;#125;&amp;#125; (content) &amp;#123;&amp;#123;end&amp;#125;&amp;#125;</code>.</p>');
```

Comment:
It seems that `&amp;#123;` is escaped twice and it's displayed as `&#123;` instead of `{` in the browser. Is this expected?

Reply:
I will update the test case using `escapeHTML` to make it clearer.

Reply:
@stevenjoezhang 

The `marked.js` will escape code by default, thus caused double escape.